src/api/procedures/modifyClaims.ts

```diff
@@ -169,6 +169,47 @@ export async function prepareModifyClaims(
       modifyClaimArgs.map(([identityId, claim]) => tuple(identityId, claim))
     );
   } else {
+    if (operation === ClaimOperation.Add) {
+      const existingCddId: string[] = [];
+
+      const claimTargets = claims.filter(
+        ({ claim: { type } }) => type === ClaimType.CustomerDueDiligence
+      );
+
+      const issuedCddClaims = await context.issuedClaims({
+        targets: claimTargets.map(({ target }) => target),
+        claimTypes: [ClaimType.CustomerDueDiligence],
+        includeExpired: false,
+      });
+
+      claimTargets.forEach(({ target, claim }) => {
+        const did = signerToString(target);
+        const filterCdds = issuedCddClaims.data.filter(
+          ({ target: issuedTarget }) => issuedTarget.did === did
+        );
+
+        if (filterCdds.length) {
+          // we know the claim is a CDD claim, so it must have an id property
+          const { id } = filterCdds[0].claim as { id: string };
+          const { id: cddId } = claim as { id: string };
+
+          if (id !== cddId) {
+            existingCddId.push(id);
+          }
+        }
+      });
+
+      if (existingCddId.length) {
+        throw new PolymeshError({
+          code: ErrorCode.ValidationError,
+          message: 'This Identity already has CDD claims with a different ID',
+          data: {
+            existingCddId: uniq(existingCddId),
+          },
+        });
+      }
```

Comment:
You need to take into account the fact that there might be multiple identities that cause the error. You should be storing occurrences in a `{ target: Identity; currentCddId: string; newCddId: string }[]` variable called `invalidCddClaims` instead of just the CDD id without its corresponding Identity.

Also, the message should be `A target Identity cannot have CDD claims with different IDs`